relayer/processor/path_processor_internal.go

```diff
@@ -590,40 +619,37 @@ func (pp *PathProcessor) assembleAndSendMessages(
 	if len(messages.packetMessages) == 0 && len(messages.connectionMessages) == 0 && len(messages.channelMessages) == 0 {
 		return nil
 	}
-	var outgoingMessages []provider.RelayerMessage
+	outgoingMessages := make(
+		[]provider.RelayerMessage,
+		0,
+		len(messages.packetMessages)+len(messages.connectionMessages)+len(messages.channelMessages),
+	)
 	msgUpdateClient, err := pp.assembleMsgUpdateClient(ctx, src, dst)
 	if err != nil {
 		return err
 	}
 	outgoingMessages = append(outgoingMessages, msgUpdateClient)
 
+	var wg sync.WaitGroup
```

Comment:
As we start using the wait group, can you leave a comment explaining why this is split into many goroutines?

I think it is because some of the message assembly will need to query the chain -- a sensible case for concurrent operations -- but it would help to have a note.